pkg/core/consensus/selection/selector.go

```diff
@@ -138,30 +150,28 @@ func (s *Selector) IncreaseTimeOut() {
 	s.timeout = s.timeout * 2
 }
 
-func (s *Selector) publishBestEvent() error {
+func (s *Selector) sendBestEvent() error {
+	var bestEvent consensus.InternalPacket
 	s.eventPlayer.Pause(s.scoreID)
-	buf := new(bytes.Buffer)
 	s.lock.RLock()
-	bestEvent := s.bestEvent
+	bestEvent = s.bestEvent
 	s.lock.RUnlock()
+
 	// If we had no best event, we should send an empty hash
 	if bestEvent == nil {
-		s.signer.SendInternally(topics.BestScore, emptyScore[:], buf, s.ID())
-	} else {
-		s.signer.SendInternally(topics.BestScore, bestEvent.VoteHash, buf, s.ID())
+		bestEvent = s.signer.Compose(emptyScoreFactory{})
 	}
 
+	msg := message.New(topics.Score, bestEvent)
```

Comment:
This message topic should be `BestScore`, as it would not arrive at the first step reducer otherwise.

Reply:
the message is sent under the `topics.BestScore` (`s.signer.SendInternally(topics.BestScore, msg, s.ID())`). However, the message category is that of a `Score` (pretty much like sending on `Gossip` any message would cause the category and topic to differ). Is this causing problems?

Reply:
Yup, as the `Coordinator` routes messages based on their `Message.Category`, and not the topic given to `SendInternally`.

Reply:
Gotcha

Reply:
I will correct it, but this seems like an architectural flaw. It forces messages to know where they will end up to and breaks message immutability. The example of `BestScore` and `Score` is quite emblematic in that respect. It would perhaps be better for the `multilistener` to reroute messages internally on the basis of their topic, rather than their category.

Reply:
That could work as well. Up to you

Reply:
I opened an issue about it as it could lead to more refactoring, which I wanna avoid within the context of this PR